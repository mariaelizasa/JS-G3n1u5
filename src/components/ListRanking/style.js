import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  color: #1f6f8b;
  font-size: 24px;

  ul {
    display: flex;
    margin: 15px 20px;
  }
  .score {
    display: flex;
    position: absolute;
    right: 0px;
    margin: 15px 20px;
  }
`;
