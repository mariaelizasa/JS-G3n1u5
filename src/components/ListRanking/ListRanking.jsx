import React from "react";
import { Container } from "./style";

const ListRanking = (props) => {
  return (
    <Container props={props.index}>
      <ul>{props.index}</ul>
      <ul data-testid="name">{props.name}</ul>
      <ul data-testid="score" className="score">
        {props.score}
      </ul>
    </Container>
  );
};

export default ListRanking;
