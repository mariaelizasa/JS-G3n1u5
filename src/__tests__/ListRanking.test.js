import React from "react";
import { render } from "@testing-library/react";

import ListRanking from "../components/ListRanking/ListRanking";

describe.only("Testing Game Component", () => {
  it("should wait for the rendered component", () => {
    const view = render(<ListRanking />);
    expect(view).toBe(view);
  });

  it("should ensure that firstchild element will return expected result from passed value in the parameter", () => {
    const { getByTestId } = render(<ListRanking name="test" />);
    const getText = getByTestId("name");
    expect(getText.firstChild).toMatchSnapshot("test");
  });

  it("should ensure that innerHTML descendant elements when receiving passed value in parameter", () => {
    const { getByTestId } = render(<ListRanking score="1" />);
    const getScore = getByTestId("score");
    expect(getScore.innerHTML).toEqual("1");
  });
});
