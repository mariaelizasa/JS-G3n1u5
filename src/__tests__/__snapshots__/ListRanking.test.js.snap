// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Game Component should ensure that firstchild element will return expected result from passed value in the parameter: test 1`] = `test`;
